mzLib/FlashLFQ/FlashLfqEngine.cs

```diff
@@ -236,6 +259,93 @@ public FlashLfqResults Run()
                 _peakIndexingEngine.ClearIndex();
             }
 
+
+            //IsoTracker
+            if (IsoTracker)
+            {
+                IsoTrackerIsRunning = true; // Turn on the flag, then we will use the separate indexEngine for each files
+                IsobaricPeakInDifferentRun = new Dictionary<string, Dictionary<Tuple<int, double, double>, List<ChromatographicPeak>>>();
+                var idGroupedBySeq = _allIdentifications.Where(p=> p.BaseSequence != p.ModifiedSequence).GroupBy(p => p.ModifiedSequence);
+                // We only consider the modified sequence in the IsoTracker
+
+                foreach (var idGroup in idGroupedBySeq) //Try to iterate the all ID in the group
+                {
+                    List<XIC> xicGroup = new List<XIC>();
+                    var mostCommonChargeIdGroup = idGroup.GroupBy(p => p.PrecursorChargeState).OrderBy(p => p.Count()).Last();
```

Comment:
This shouldn't be in the main Run method. You need to create a separate method where IsoTracker is run